var/spack/repos/builtin/packages/hepmcanalysis/package.py

```diff
@@ -22,17 +22,17 @@ class Hepmcanalysis(MakefilePackage):
     depends_on('root')
     depends_on('clhep')
 
-    variant('cxxstd',
-            default='11',
-            values=('11', '14', '17'),
-            multi=False,
-            description='Use the specified C++ standard when building.')
-
     patch('lcg.patch')
 
+    @run_before('build')
+    def patch(self):
+        filter_file("TDirectory::CurrentDirectory()->",
+                    "TDirectory::CurrentDirectory().",
+                    "src/baseAnalysis.cc")
+
     def edit(self, spec, prefix):
```

Comment:
This might need  a `before_build`, but  can just be added to `patch()`

Reply:
It already runs before build:
```
==> Applied patch /cvmfs/gitlab-runner/VHY8orns/0/sft/sft-spack-repo/spack/var/spack/repos/builtin/packages/hepmcanalysis/lcg.patch
==> Ran patch() for hepmcanalysis
==> hepmcanalysis: Executing phase: 'edit'
==> hepmcanalysis: Executing phase: 'build'
==> Error: ProcessError: Command exited with status 2:
    'make' '-j4'
```

Reply:
But yes, I could add it to patch anyway